src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/crud-form.component.ts

```diff
@@ -32,28 +37,46 @@ export class CrudFormComponent implements OnInit {
     private dataGatewayService: DataGatewayService,
     private activatedRoute: ActivatedRoute,
     private taskWrapper: TaskWrapperService,
-    private location: Location
+    private location: Location,
+    private router: Router
   ) {}
 
   ngOnInit(): void {
     this.activatedRoute.data.subscribe((data: any) => {
       this.resource = data.resource;
       this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => {
-        this.title = response.forms[0].control_schema.title;
-        this.uiSchema = response.forms[0].ui_schema;
-        this.controlSchema = response.forms[0].control_schema;
+        for (let i = 0; i < response.forms.length; i++) {
+          if (this.router.url.includes(response.actions[i].routerLink)) {
+            this.urlFormName = this.router.url.substring(this.router.url.lastIndexOf('/') + 1);
+            this.methodType = response.forms[i].method_type;
+            this.title = response.forms[i].container_schema.control_schema.title;
+            this.uiSchema = response.forms[i].container_schema.ui_schema;
+            this.controlSchema = response.forms[i].container_schema.control_schema;
+          }
+        }
       });
     });
+    if (this.router.url.includes('/edit')) {
```

Comment:
Instead of making the component change its behaviour based on the URL, isn't it possible to tell the component via `@Input`s? "Pure" classes (whose behaviour only depends on their constructor arguments) are ideal since they are easier to implement, debug, unit-test, etc.